src/index.tsx

```diff
@@ -11,117 +11,160 @@ import Animated, {
   useSharedValue,
 } from 'react-native-reanimated';
 
+type MarqueeDirection = 'up' | 'left' | 'right' | 'down';
+
 const AnimatedChild = ({
   index,
   children,
   anim,
+  textHeight,
   textWidth,
   spacing,
+  direction = 'right',
 }: React.PropsWithChildren<{
   index: number;
   anim: SharedValue<number>;
+  textHeight: SharedValue<number>;
   textWidth: SharedValue<number>;
   spacing: number;
+  direction?: MarqueeDirection;
 }>) => {
   const stylez = useAnimatedStyle(() => {
+    const vertical = direction === 'up' || direction === 'down';
+    const horizontal = direction === 'left' || direction === 'right';
+
     return {
       position: 'absolute',
-      left: index * (textWidth.value + spacing),
+      top: vertical && direction === 'up' ? index * (textHeight.value + spacing) : undefined,
+      bottom: vertical && direction === 'down' ? index * (textHeight.value + spacing) : undefined,
+      right: horizontal && direction === 'right' ? index * (textWidth.value + spacing) : undefined,
+      left: horizontal && direction === 'left' ? index * (textWidth.value + spacing) : undefined,
       transform: [
         {
-          translateX: -(anim.value % (textWidth.value + spacing)),
+          ...(vertical
+            ? { translateY: (anim.value % (textHeight.value + spacing)) * (direction === 'up' ? -1 : 1) }
+            : { translateX: (anim.value % (textWidth.value + spacing)) * (direction === 'left' ? -1 : 1) }),
         },
       ],
     };
-  }, [index, spacing, textWidth]);
+  }, [index, spacing, textHeight, textWidth]);
   return <Animated.View style={stylez}>{children}</Animated.View>;
 };
 
 export type MarqueeProps = React.PropsWithChildren<{
   speed?: number;
   spacing?: number;
   style?: ViewStyle;
+  /**
+   * Direction in which the marquee should scroll
+   */
+  direction?: MarqueeDirection;
+  /**
+   * Delay in milliseconds before the marquee starts
+   */
+  delay?: number;
 }>;
 
 /**
  * Used to animate the given children in a horizontal manner.
  */
-export const Marquee = React.memo(
-  ({ speed = 1, children, spacing = 0, style }: MarqueeProps) => {
-    const parentWidth = useSharedValue(0);
-    const textWidth = useSharedValue(0);
-    const [cloneTimes, setCloneTimes] = React.useState(0);
-    const anim = useSharedValue(0);
+const Marquee = React.memo(({ speed = 1, children, spacing = 0, style, delay, direction = 'right' }: MarqueeProps) => {
+  const parentHeight = useSharedValue(0);
```

Comment:
I propose to store both `width` and `height` under the same `SharedValue` (same for `text` sizes).

E.g:

```jsx
const parentSize = useSharedValue({width: 0, height: 0});
const textSize = useSharedValue(width: 0, height: 0});
```